### PR TITLE
Fixed a bug where materials and/or meshes weren't assigned to scene on first import.

### PR DESCRIPTION
**Pull request by NathanWarden**
Fixes https://github.com/godotengine/godot/issues/11452

### CODE REVIEW

File: editor/import/resource_importer_scene.cpp [hunk 1]
```diff
@@ -858,12 +858,11 @@ void ResourceImporterScene::_make_external_resources(Node *p_node, const String
...
 					if (p_keep_materials && FileAccess::exists(ext_name)) {
 						//if exists, use it
-						Ref<Material> existing = ResourceLoader::load(ext_name);
-						p_materials[mat] = existing;
+						p_materials[mat] = ResourceLoader::load(ext_name);
 					} else {
 
 						ResourceSaver::save(ext_name, mat, ResourceSaver::FLAG_CHANGE_PATH);
-						p_materials[mat] = mat;
+						p_materials[mat] = ResourceLoader::load(ext_name);
```
Comment (akien-mga) [inline editor/import/resource_importer_scene.cpp]: What's the reason for this change, and the other similar ones (reloading the just-saved mat instead of just using the source one)?

Comment (NathanWarden) [reply editor/import/resource_importer_scene.cpp]: It's because if it uses the original sourced one the resulting imported mesh or scene uses a builtin material instead of the saved resource even though the import setting is set to "Files" on the initial import. I have a video of the issue in the linked bug report.

In other words, the original material is not the same material as the newly saved resource. Thus, when you go to make changes it won't overwrite the saved material even though the import setting is set to "Files", it will try and create another new one. The same issue exists with meshes that this PR fixes.

**Comment by NathanWarden** [comment]
Before the beta hits I just want to make sure that it's clear what this fixes. I can make a video showing before and after. Actually, there's a video that I made that's in the bug report. But, I can make a video showing it after the fix. Thanks! :)
